### PR TITLE
Updated the event schema

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

Updated the event schema so that it is more feasible to recreate the state using events.

# Review Checklists

Please check each item **before approving** the pull request. While going
through the checklist, it is recommended to leave comments on items that are
referenced in the checklist to make sure that they are reviewed. If there are
multiple reviewers, copy the checklists into sections titled `## [Reviewer Name]`.
If the PR doesn't touch Solidity and/or Rust, the corresponding checklist can
be removed.

## [[Reviewer Name]]

- [ ] **Tokens**
    - [ ] Do all `approve` calls use `forceApprove`?
    - [ ] Do all `transfer` calls use `safeTransfer`?
    - [ ] Do all `transferFrom` calls use `msg.sender` as the `from` address?
        - [ ] If not, is the function access restricted to prevent unauthorized
              token spend?
- [ ] **Low-level calls (`call`, `delegatecall`, `staticcall`, `transfer`, `send`)**
    - [ ] Is the returned `success` boolean checked to handle failed calls?
    - [ ] If using `delegatecall`, are there strict access controls on the
          addresses that can be called? It shouldn't be possible to `delegatecall`
          arbitrary addresses, so the list of possible targets should either be
          immutable or tightly controlled by an admin.
- [ ] **Reentrancy**
    - [ ] Are functions that make external calls or transfer ether marked as `nonReentrant`?
        - [ ] If not, is there documentation that explains why reentrancy is
              not a concern or how it's mitigated?
- [ ] **Gas Optimizations**
    - [ ] Is the logic as simple as possible?
    - [ ] Are the storage values that are used repeatedly cached in stack or
          memory variables?
    - [ ] If loops are used, are there guards in place to avoid out-of-gas
          issues?
- [ ] **Visibility**
    - [ ] Are all `payable` functions restricted to avoid stuck ether?
- [ ] **Math**
    - [ ] Is all of the arithmetic checked or guarded by if-statements that will
          catch underflows?
    - [ ] If `Safe` functions are altered, are potential underflows and
          overflows caught so that a failure flag can be thrown?
    - [ ] Are all of the rounding directions clearly documented?
- [ ] **Testing**
    - [ ] Are there new or updated unit or integration tests?
    - [ ] Do the tests cover the happy paths?
    - [ ] Do the tests cover the unhappy paths?
    - [ ] Are there an adequate number of fuzz tests to ensure that we are
          covering the full input space?

### CODE REVIEW

File: contracts/src/internal/HyperdriveLP.sol [hunk 1]
```diff
@@ -383,12 +373,8 @@ abstract contract HyperdriveLP is
             msg.sender, // provider
             _options.destination, // destination
             _lpShares,
-            _convertToBaseFromOption(proceeds, vaultSharePrice, _options), // base proceeds
-            _convertToVaultSharesFromOption(
```
Comment (jrhea) [inline contracts/src/internal/HyperdriveLP.sol:387]: Can you get rid of these helper methods now?

**Review by jrhea — COMMENTED** [review]
This seems like a good simplification and should reduce gas costs some. A few comments:

1) did the coverage go down bc the loc was reduced?
2) is the frontend team aware that they will have to look at the asBase option to determine the units?
3) can we get rid of the convertToBase/SharesFromOption helper methods?

File: contracts/src/internal/HyperdriveLP.sol (hunk 1 above)
Comment (jalextowle) [reply contracts/src/internal/HyperdriveLP.sol:387]: Yep, I did that

File: test/units/hyperdrive/FeeTest.t.sol [hunk 2]
```diff
@@ -1,7 +1,11 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity 0.8.20;
 
+// FIXME
+import { console2 as console } from "forge-std/console2.sol";
```
Comment (jrhea) [inline test/units/hyperdrive/FeeTest.t.sol:5]: fix me

**Review by jrhea — APPROVED** [review]
looks good!

**Pull request merged**